src/api/cluster_create.go

```diff
@@ -110,6 +110,8 @@ func (a *ClusterAPI) CreateCluster(c *gin.Context) {
 			createClusterRequest.SecretId = secret.GenerateSecretIDFromName(createClusterRequest.SecretName)
 		}
 
+		createClusterRequest.ProductDetailsGetter = a.productDetailsGetter
```

Comment:
**Question**: would it be much more complex if instead of proxying the product details getter here, we would retrieve the details (below the existing cluster checking part) and pass the static details to the validation function?

Reply:
I think we can only do it at the nodepool level because the detail getter method needs the exact instance type, zone and spotPrice and this can be differ between node pools within the same cluster.